ledger/persistedaccts_list.go

```diff
@@ -0,0 +1,127 @@
+package ledger
+
+// persistedAccountDataList represents a doubly linked list.
+// must initiate with newPersistedAccountList.
+type persistedAccountDataList struct {
+	root     persistedAccountDataListNode // sentinel list element, only &root, root.prev, and root.next are used
+	len      int                          // current list length excluding (this) sentinel element
```

Comment:
I can't see any real good use of `len`. The only "usage" of it is to determine if we have any entries in the list, and we can do that by comparing `root.next == root.prev`

Reply:
i've removed the len property. The inspection for 0 length is now: `list.root.next == &list.root`.